File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/fragments/AppIndexFragment.java
package com.inspiredandroid.linuxcommandbibliotheca.fragments;

import android.os.Bundle;

import com.google.android.gms.appindexing.Action;
import com.google.android.gms.appindexing.AppIndex;
import com.google.android.gms.common.api.GoogleApiClient;

/**
 * Created by simon on 20.11.15.
 */
public abstract class AppIndexFragment extends SuperFragment {

    private GoogleApiClient mClient;

    /**
     * Title for current page, shown in autocompletion UI
     *
     * @return
     */
    public abstract String getAppIndexingTitle();

    /**
     * Construct the Action performed by the user
     *
     * @return
     */
    public abstract Action getAppIndexingAction();

    @Override
    public void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);

        mClient = new GoogleApiClient.Builder(getContext()).addApi(AppIndex.API).build();
    }

    @Override
    public void onStart()
    {
        super.onStart();

        mClient.connect();
        AppIndex.AppIndexApi.start(mClient, getAppIndexingAction());
    }

    @Override
    public void onStop()
    {

        AppIndex.AppIndexApi.end(mClient, getAppIndexingAction());
        mClient.disconnect();

        super.onStop();
    }

}

File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/fragments/AppIndexFragment.kt
package com.inspiredandroid.linuxcommandbibliotheca.fragments

import android.support.v4.app.Fragment
import com.google.firebase.appindexing.Action
import com.google.firebase.appindexing.FirebaseUserActions

/* Copyright 2019 Simon Schubert
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/
abstract class AppIndexFragment : Fragment() {

    /**
     * Construct the Action performed by the user
     *
     * @return
     */
    abstract fun getAppIndexingAction(): Action

    override fun onStart() {
        super.onStart()
        FirebaseUserActions.getInstance().start(getAppIndexingAction())
    }

    override fun onStop() {
        FirebaseUserActions.getInstance().end(getAppIndexingAction())
        super.onStop()
    }

}
